src/fsharp/NameResolution.fs

```diff
@@ -4320,11 +4320,12 @@ let rec ResolvePartialLongIdentPrim (ncenv: NameResolver) (nenv: NameResolutionE
            | FullyQualified -> []
            | OpenQualified ->
                nenv.eUnqualifiedItems.Values
-               |> List.filter (function
+               |> Seq.filter (function
                    | Item.UnqualifiedType _ -> false
                    | Item.Value v -> not v.IsMember
                    | _ -> true)
-               |> List.filter (ItemIsUnseen ad g ncenv.amap m >> not)
+               |> Seq.filter (ItemIsUnseen ad g ncenv.amap m >> not)
+               |> Seq.toList
```

Comment:
Why this change was needed?

Reply:
I doubt it was necessary

Reply:
There was an internal implementation of keys/values which returned a list and the implementation was counting with that.

Since the new public implementation of keys/values is an ICollection I had to change this to take seq instead of List otherwise it would not compile.